smarts/core/agent.py

```diff
@@ -58,6 +58,15 @@ def act(self, obs):
 
         raise NotImplementedError
 
+    def step(self, state, action, reward, next_state, done, info):
+        return {}
```

Comment:
if the goal is to have users override these methods maybe have `raise NotImplementedError`?

Reply:
I think for these methods, a default implementation may be needed. These are more for RL agents to be able to save/load from a model, and also to step given knowledge of where an action in a state lead. (Sorry I should have specified this in the description! Will update the description.)

I am unsure if this sort of implementation is what the SMARTS team is wanting, but these methods (or similar) are what ULTRA would need.